edgelet/edgelet-docker/src/image_prune_data.rs

```diff
@@ -0,0 +1,646 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use std::io::Write;
+use std::path::{Path, PathBuf};
+use std::sync::{Arc, Mutex};
+use std::time::UNIX_EPOCH;
+use std::{collections::HashMap, collections::HashSet, fs, time::Duration};
+
+use edgelet_settings::base::image::ImagePruneSettings;
+
+use crate::Error;
+
+const IMAGE_USE_FILENAME: &str = "image_use";
+const TMP_FILENAME: &str = "image_use_tmp";
+
+const DEFAULT_CLEANUP_TIME: &str = "00:00"; // midnight
+const DEFAULT_RECURRENCE_IN_SECS: u64 = 60 * 60 * 24; // 1 day
+const DEFAULT_MIN_AGE_IN_SECS: u64 = 60 * 60 * 24 * 7; // 7 days
+
+#[derive(Debug, Clone)]
+struct ImagePruneInner {
+    image_use_filepath: String, // Copyright (c) Microsoft. All rights reserved.
+    tmp_filepath: String,
+    settings: ImagePruneSettings,
+}
+
+#[derive(Debug, Clone)]
+pub struct ImagePruneData {
```

Comment:
If the official name of this feature is Image Garbage Collection, I suggest changing this struct to `ImageGarbageCollectionData`